orderbook/src/orderbook.rs

```diff
@@ -51,8 +51,8 @@ pub struct Orderbook {
     event_updater: Mutex<EventUpdater>,
     balance_fetcher: Box<dyn BalanceFetching>,
     fee_validator: Arc<EthAwareMinFeeCalculator>,
-    unsupported_tokens: HashSet<H160>,
     min_order_validity_period: Duration,
+    bad_token: Arc<dyn BadTokenDetecting>,
```

Comment:
naming nit: calling this bad_token sounds like this is a single H160. Maybe instead call it bad_token_detector or at least bad_tokens?